lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -194,23 +193,23 @@ public function getListTableForeignKeysSQL($table, $database = null)
     /**
      * {@inheritDoc}
      */
-    public function getCreateViewSQL($name, $sql)
+    public function getCreateViewSQL(string $name, string $sql) : string
     {
         return 'CREATE VIEW ' . $name . ' AS ' . $sql;
     }
 
     /**
      * {@inheritDoc}
      */
-    public function getDropViewSQL($name)
+    public function getDropViewSQL(string $name) : string
     {
         return 'DROP VIEW ' . $name;
     }
 
     /**
      * {@inheritDoc}
      */
-    protected function getVarcharTypeDeclarationSQLSnippet($length, $fixed)
+    protected function getVarcharTypeDeclarationSQLSnippet(?int $length, bool $fixed) : string
```

Comment:
Wouldn't `0` be better default?

Reply:
There's no default. It's a nullable length for platforms like SQLite which don't support length or PostgreSQL which don't require one.

I have some PoC changes in https://github.com/morozov/dbal/commits/remove-default-max-length.